### DISCUSSION TITLE
How to implement static page generation if my API server is not available at the Next build stage?

### DISCUSSION THREAD

**Discussion by maksim-kononov-csssr** [vercel/next.js#52794] [App Router]
My infrastructure is configured in a such way that my frontend and backend are building in parallel. During the Next build stage my API server does not work - it cannot respond to requests with data, so Next caches the "error" response, as a result my website displays stubs instead of data. The only way I found to update the "error cache" is to make an additional request to revalidate data. But this is a rather fragile workaround to my taste, because every time after the Next start I have to make additional requests for revalidation. Is there any better way to set up this process? I think these are quite common conditions when we cannot guarantee that at the Next build stage we have a working API server.

**Comment by icyJoseph** [comment]
I mean, this is clearly a catch-22 in the build system you are using. 

Could you deploy sequentially instead? If you want to statically generate pages with an input source that doesn't exist at build time, then you need to wait until it exist.

**Reply by maksim-kononov-csssr** [reply]
Yes, I thought about changing the build system, but before that, I want to make sure that this is really the only way.

**Reply by icyJoseph** [reply]
well, unless you could connect to the data provider. Say if the backend is reading data from a db, maybe you could connect to that during build time? Otherwise, doing it sequentially should be fine, I guess.
